### PR TITLE
[New Template] NP-AA1996-MAIN and include organization for CET transcript

### PR DESCRIPTION
**Pull request by mmysgp**
…anscript

---
name: Certificate Template Addition
about: This is the workflow for requesting new certificate templates to be added to
  the OpenCerts repository
title: "[New Template]"
labels: new template
assignees: ''

---

# Pull Request Guidelines for Adding Certificate Templates
This document is a work in progress but here are some basic checks. As these are only basic guidelines, meeting the below doesn't indicate there will be no issues with your pull request.

### Certificate Template 
- [x] Integration test for each template that checks that the correct rendering is done given a sample certificate
- [x] Sample certificate file included for each template, located alongside the integration test for each template
- [x] Sample certificates must obviously be a sample certificate
  - [x] Obviously fictitious name
  - [x] Obviously sample signatory images
- [x] No fixed-size raster images as part of certificate layout
- [ ] Mobile responsive design
- [x] Date parsing should be localised to template author's timezone
- [x] Webpack chunking code is correct
  - [x] Has chunking code
  - [x] Same chunking code as the other certificates belonging to that institute
- [ ] Certificate Store Addresses have been updated
  - [ ] Template Whitelist
  - [ ] Registry
- [x] Template should not be using resources(images etc.) on the website outside of their own folder (e.g institute logo shouldn't be used from /static because there's no guarantee it will not change)

### Pre-merge checks

- [x] Ensure that your code has been [rebased](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) on top of latest OpenCerts master
- [x] Linter issues resolved (Run `npm run lint:fix` to see issues)
- [x] `npm run test` passes
- [x] `npm run test:integration` passes
- [x] [Travis Build passes](https://docs.travis-ci.com/user/for-beginners/)

### CODE REVIEW

**Comment by mmysgp** [comment]
Hi Raymond & Ruijie,

Please help review this pull request and merge the changes to master.
Thanks.

**Comment by mmysgp** [comment]
Hi Raymond & Ruijie,

Suggest to update the instruction in "Customization Template" to use "**npm ci**", instead of "**npm install**", otherwise, package-lock.json will be modified everytime after the installation, which results in json conflict when a pull request is submitted.

**Comment by rjchow** [comment]
hmmm ideally the package.lock shouldn't be updated if no packages were added but i'll have a look at it

File: src/components/CertificateTemplates/tlds/sg/edu/np/NP-AA1996-MAIN/NP_Certs_FT_MAIN_1996.opencert [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        "email": "ba912622-882a-4245-bfb7-27f28068cbca:string:askNP@np.edu.sg",
+        "certificateStore": "4374fbe3-d4f6-4c8e-a87d-7f70faa4e838:string:0xc36484efa1544c32ffed2e80a1ea9f0dfc517495"
+      }
+    ],
+    "recipient": {
+      "name": "056285f0-c0d3-4f35-9a10-bf5cd6856b2e:string:Student Name MAIN Cert 1996",
+      "nric": "213a4ec1-90c8-4cf4-ae81-a1aad885c38d:string:S1234567A"
+    },
+    "additionalData": {
+      "studentId": "edd7ebf0-6298-4405-9dda-3c35a6407f15:string:12345678A",
```
Comment (rjchow) [inline src/components/CertificateTemplates/tlds/sg/edu/np/NP-AA1996-MAIN/NP_Certs_FT_MAIN_1996.opencert]: we've moved studentId to recipient object, up to you if you want to move yours there too - ok either way

File: src/components/CertificateTemplates/tlds/sg/edu/np/NP-CET2019-PDPMAIN/NP_Certs_PDP_MAIN_2019.opencert [hunk 2]
```diff
@@ -1,217 +1,218 @@
 {
-  "schema": "opencerts/v1.4",
+  "schema": "opencerts/v1.5",
```
Comment (rjchow) [inline src/components/CertificateTemplates/tlds/sg/edu/np/NP-CET2019-PDPMAIN/NP_Certs_PDP_MAIN_2019.opencert]: intended?

File: src/components/CertificateTemplates/tlds/sg/edu/np/NP-CET2019-PTDMAIN/NP_Certs_PTD_MAIN_2019.opencert [hunk 3]
```diff
@@ -1,387 +1,388 @@
 {
-  "schema": "opencerts/v1.4",
+  "schema": "opencerts/v1.5",
```
Comment (rjchow) [inline src/components/CertificateTemplates/tlds/sg/edu/np/NP-CET2019-PTDMAIN/NP_Certs_PTD_MAIN_2019.opencert]: intended?

File: src/components/CertificateTemplates/tlds/sg/edu/np/NP-CET2019-SDPCN/NP_Certs_PDP_SDPCN_2019.opencert [hunk 4]
```diff
@@ -1,136 +1,137 @@
 {
-  "schema": "opencerts/v1.4",
+  "schema": "opencerts/v1.5",
```
Comment (rjchow) [inline src/components/CertificateTemplates/tlds/sg/edu/np/NP-CET2019-SDPCN/NP_Certs_PDP_SDPCN_2019.opencert]: ^

Comment (mmysgp) [reply src/components/CertificateTemplates/tlds/sg/edu/np/NP-CET2019-SDPCN/NP_Certs_PDP_SDPCN_2019.opencert]: will use 1.4.

File: src/components/CertificateTemplates/tlds/sg/edu/np/NP-CET2019-PTDMAIN/NP_Certs_PTD_MAIN_2019.opencert (hunk 3 above)
Comment (mmysgp) [reply src/components/CertificateTemplates/tlds/sg/edu/np/NP-CET2019-PTDMAIN/NP_Certs_PTD_MAIN_2019.opencert]: will use 1.4.

File: src/components/CertificateTemplates/tlds/sg/edu/np/NP-CET2019-PDPMAIN/NP_Certs_PDP_MAIN_2019.opencert (hunk 2 above)
Comment (mmysgp) [reply src/components/CertificateTemplates/tlds/sg/edu/np/NP-CET2019-PDPMAIN/NP_Certs_PDP_MAIN_2019.opencert]: will use 1.4.

File: src/components/CertificateTemplates/tlds/sg/edu/np/NP-AA1996-MAIN/NP_Certs_FT_MAIN_1996.opencert (hunk 1 above)
Comment (mmysgp) [reply src/components/CertificateTemplates/tlds/sg/edu/np/NP-AA1996-MAIN/NP_Certs_FT_MAIN_1996.opencert]: will keep it as is.

**Comment by mmysgp** [comment]
> hmmm ideally the package.lock shouldn't be updated if no packages were added but i'll have a look at it

I found inexplicit version in package-lock.json. Not sure whether it causes the problem. 
image
By right, package-lock.json always indicate a specific version number.

**Pull request merged**